.github/workflows/ci.yml

```diff
@@ -143,9 +143,9 @@ jobs:
 
       - name: Clang Tidy
         run: |
-          export MODIFIED_FILES=$(git diff -U0 HEAD^ --name-only | grep -E "\.(cpp|hpp|hxx)$")
+          export MODIFIED_FILES=$(git diff --name-only --diff-filter=d origin/${{ github.base_ref }} origin/${{ github.head_ref }} | grep -E "\.(cpp|hpp|hxx)$")
```

Comment:
These variables cannot been accessed on pull events. This is the reason why we decide to extract this build into a separate file.

In this PR, keep only the fix that concerns the path of the workspace